onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -2246,7 +2246,7 @@
   // For in-place reuse tensors, the lifetime is the union of all the tensors that tensors that use that buffer
 #if !defined(ORT_MINIMAL_BUILD) && defined(ORT_MEMORY_PROFILE)
   void AdjustInplaceLifeIntervals() {
-    InlinedHashMap<OrtValueIndex, std::vector<OrtValueIndex>> inplace_reuse_buffer;
+    InlinedHashMap<OrtValueIndex, InlinedVector<OrtValueIndex>> inplace_reuse_buffer;
```

Comment:
>InlinedHashMap<OrtValueIndex, InlinedVector<OrtValueIndex>> inplace_reuse_buffer; [](http://example.com/codeflow?start=4&length=81)

Use reserve(), otherwise, it becomes worse than with std::vector.